src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
IMO we should not fake the plugin instance. If the plugin has no instance or we cannot obtain it we should simply tell the developer so using `Optional.empty()` then he can decide whether he uses the container instance or maybe utilize another API to fetch the plugin instance.

````java
container.getInstance().orElse(container);
````
or
````java
container.getInstance().orElseGet(() -> magicAPI.getPluginInstance(container.getId()));
````